system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp

```diff
@@ -192,8 +192,8 @@ static void submitKQLQuery(std::string & readBuffer, const char * token, const c
         char                    curlErrBuffer[CURL_ERROR_SIZE];
         curlErrBuffer[0] = '\0';
 
-        char * encodedKQL = curl_easy_escape(curlHandle, kql, strlen(kql));
-        char * encodedTimeSpan = curl_easy_escape(curlHandle, timeSpan, strlen(timeSpan));
+        char * encodedKQL = curl_easy_escape(curlHandle, kql, strlen(kql)+1);
```

Comment:
From the documentation I don't think this change is correct.  It should be the length of the string - this will add a %00 onto the end which you don't want.
I think coverity is actually complaining about a bug in the curl code - where the maximum length is not including a byte for the null terminator.  I think
curl_easy_escape(, "\0", 1) may return CURLE_TOO_LARGE ( Ihaven't tested).